### DISCUSSION TITLE
Version 3.0.0

### DISCUSSION THREAD

**Discussion by gustafandersson-cubist** [dotnet-bluetooth-le/dotnet-bluetooth-le#695] [General]
Is there a roadmap for releasing 3.0.0 as a stable release?

**Comment by janusw** [comment]
I would like to release 3.0.0 over the summer, in a couple of weeks if all goes well. The betas should be pretty stable already. Do they work well for you? Do you have any comments?

**Reply by gustafandersson-cubist** [reply]
They do. Very appreciated!

**Reply by bogardani** [reply]
Just my two cents for the insanely dedicated people who decided to use the Android Ble Api.

So far we had two bugs due to the incosistency of the Android Api. I emphasize that these are not related to this project, I just want you to be aware of them.

Our project only supports the latest Api (Api 33, Android 13). It requires us to connect to peripherals in rapid succession and in parallel, and do operations on them.
The first bug is related to this ticket: [https://issuetracker.google.com/issues/269646835](url) the notorious Gatt 133 error. I could perfectly reproduce it on my device (Samsung Galaxy A53 5G), but the ticket mentions several other devices. We had to drop support for these, and probably more in the future, as setting the peripheral's advertising interval under 600ms is not a proper solution due to environmental conditions affecting the whole advertisement concept.

The second one we could only reproduce reliably on a OnePlus 8T. When you connect to the same peripheral multiple times (with proper disconnect of course) and do some characteristic read/write during the connection, if you don't immediately call device.GetServices() when establishing a connection, right after the DeviceConnected callback finishes. The later called device.GetServiceAsync (which calls the internal discoverServices) somehow makes the Android Api reconnect the peripheral, going through the whole callback chain, invoking OnConnectionStateChange with status = 0 newState = 2, and running DeviceConnected again. This wouldn't be a problem, but during this reconnection the originally called device.GetServiceAsync just hangs until cancelled, it doesn't matter that the Api gives back the same Gatt object. As I said, the workaround is to immediately call device.GetServices after the connection is up.

All in all I really appreciate this project, keep up the good work!

**Comment by janusw** [comment]
We're a bit behind the planned schedule, but I just published an rc, so we should be rather close to a final version 3.0.0.

**Reply by janusw** [reply]
FYI: 3.0.0 has been released today! https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/releases
